extensions/wikia/Email/Controller/FounderController.class.php

```diff
@@ -244,3 +244,89 @@ protected function getFooterEncouragement() {
 			->parse();
 	}
 }
+
+class FounderNewMemberController extends FounderController {
+	const TRACKING_CATEGORY_EN = TrackingCategories::FOUNDER_NEW_MEMBER_EN;
+	const TRACKING_CATEGORY_INT = TrackingCategories::FOUNDER_NEW_MEMBER_INT;
+
+	public function initEmail() {
+		// noop
+	}
+
+	/**
+	 * @template avatarLayout
+	 */
+	public function body() {
+		$this->response->setData( [
+			'salutation' => $this->getSalutation(),
+			'editorProfilePage' => $this->getCurrentProfilePage(),
+			'editorUserName' => $this->getCurrentUserName(),
+			'editorAvatarURL' => $this->getCurrentAvatarURL(),
+			'summary' => $this->getSummary(),
+			'buttonText' => $this->getButtonText(),
+			'buttonLink' => $this->getButtonLink(),
+			'contentFooterMessages' => [
+				$this->getWikiaCommunitySignature()
+			],
+			'hasContentFooterMessages' => true,
+			'details' => $this->getDetails(),
+		] );
+	}
+
+	public function getSubject() {
+		return $this->getMessage( 'emailext-founder-new-member-subject', $this->currentUser->getName() )->parse();
+	}
+
+	// Same message use for subject and summary
+	public function getSummary() {
+		return $this->getSubject();
+	}
+
+	public function getDetails() {
+		return $this->getMessage( 'emailext-founder-new-member-details', $this->currentUser->getName() )->parse();
+	}
+
+	public function getWikiaCommunitySignature() {
+		return $this->getMessage( 'emailext-emailconfirmation-community-team' )->text();
+	}
+
+	public function getButtonText() {
+		return $this->getMessage( 'emailext-founder-new-member-link-label' )->text();
+	}
+
+	public function getButtonLink() {
+		return $this->currentUser->getTalkPage()->getFullURL();
+	}
+
+	public function assertCanEmail() {
+		parent::assertCanEmail();
+		$this->assertFounderSubscribedToDigest();
+		$this->assertFounderWantsNewMembersEmail();
+	}
+
+	/**
+	 * If the founder is subscribed to the founder's digest, don't send them an individual email informing them
+	 * a new user joined their wiki. They'll learn about that in the digest.
+	 * @throws \Email\Check
+	 */
+	public function assertFounderSubscribedToDigest() {
+		$wikiId = \F::app()->wg->CityId;
+		if ( $this->targetUser->getBoolOption( "founderemails-complete-digest-$wikiId" ) ) {
+			throw new Check( 'Digest mode is enabled, do not create user registration event notifications' );
+		}
+	}
+
+	/**
+	 * @throws \Email\Check
+	 */
+	public function assertFounderWantsNewMembersEmail() {
+		$wikiId = \F::app()->wg->CityId;
+		if ( !$this->targetUser->getBoolOption( "founderemails-joins-$wikiId"  ) ) {
+			throw new Check( "Founder doesn't want to be emailed about new members joining this wiki" );
+		}
+	}
+
+	protected function getFooterMessages() {
```

Comment:
Ah, good catch @garthwebb. I checked with Matt and he just created a separate class for his work, he didn't end up breaking apart the `FounderController` class. I think I'll just make another noop method since it's fast and easy. If we end up needing to extend the `FounderController` class more than we can break it up, but I think after mine and Matt's ticket we should be done with the Founder emails.